stdlib/auth/src/main/ballerina/auth/Module.md

```diff
@@ -18,3 +24,21 @@ A user is denoted by a section in the configuration file. The password and the s
  password="<password>"
  scopes="<comma_separated_scopes>"
  ```
+### Outbound Authentication Provider
+
+An outbound authentication provider defines an authentication scheme that could be used to authenticate with external endpoints. The `auth:OutboundAuthProvider` acts as the interface for all the outbound authentication providers. Any type of implementation such as JDBC, JWT, OAuth2 and file-based should be object-equivalent.
+
+When creating a new outbound authentication provider, you need to implement the below functions.
+- `generateToken` : Generate the token for authentication with outbound auth providers such as JWT, OAuth2.
+- `inspect` : Inspect the incoming data and generate the token for authentication as needed. As an example if the incoming data says it needs to regenerate the token because of the previously generated token is invalid, this method will generate it.
```

Comment:
```suggestion
- `inspect`: Inspects the incoming data and generates the token for authentication as needed. For example, if the incoming data indicates that it needs to regenerate the token because the previously-generated token is invalid, this method will generate it.
```